code/game/objects/items/devices/lighting/marshalling wand.dm

```diff
@@ -0,0 +1,36 @@
+/obj/item/device/flashlight/marshallingwand
+	name = "marshalling wand"
+	desc = "A marshalling wand, used to direct air or space faring vessels during takeoff and landing."
+	icon_state = "marshallingwand"
+	item_state = "marshallingwand"
+	w_class = ITEMSIZE_SMALL
+	light_color = LIGHT_COLOR_RED
+	light_wedge = LIGHT_OMNI
+	brightness_on = 2
+	action_button_name = "Toggle Marshalling Wands"
+
+/obj/item/device/flashlight/marshallingwand/AltClick(mob/user)
+	if(!isturf(user.loc))
+		to_chat(user, "You cannot turn the light on while in this [user.loc].") //To prevent some lighting anomalities.)
+		return FALSE
+	toggle()
+	user.update_action_buttons()
+	return TRUE
+
+/obj/item/device/flashlight/marshallingwand/ui_action_click()
+	AltClick(usr)
+
+/obj/item/device/flashlight/marshallingwand/attack_self(mob/user)
+	if(!on)
+		to_chat(user, SPAN_WARNING("The wands need to be on before you can direct craft with them!"))
+		return
+	switch(user.a_intent)
+		if(I_HELP)
+			user.visible_message(SPAN_NOTICE("[user] points the marshalling wand forward, directing a pilot to take off."), SPAN_NOTICE("You point the marshalling wand forward, directing a pilot to take off."))
```

Comment:
Is it worth adding a cooldown for these?

Reply:
Possibly? I considered it, and am more than happy to do so if needed, but other things like patting animals which have a visible message don't have it